load_config.py

```diff
@@ -0,0 +1,102 @@
+# Execute the script from the root folder of the configuration you wish to
+# upload.
+
+# designed for UNIX file systems - change to c:\TEMP for Windows systems
+ZIPFILE = '/tmp/tenant.zip'
+base_url = 'https://api.sandbox.socotra.com'
+
+
+# Also supports environment variables SOCOTRA_USERNAME and SOCOTRA_PASSWORD
+# instead of CLI arguments
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--tenant_name', '-t',
+                        help='The suffix of the tenant', required=True)
+    parser.add_argument('--username', '-u',
+                        help='Socotra Config Studio username',
+                        required=False)
+    parser.add_argument('--password', '-p',
+                        help='Socotra Config Studio password',
+                        required=False)
+    parser.add_argument(
+        '--debug', '-d', help="prints debugging info", action='store_true')
+    return parser.parse_args()
+
+
+def create_zip_from_assets():
+    with zipfile.ZipFile(ZIPFILE, 'w', zipfile.ZIP_DEFLATED) as zip_file:
+        dir_path = os.getcwd()
+        for root, dirs, files in os.walk(dir_path):
+            for file in files:
+                if not file.endswith('.zip'):
+                    file_path = os.path.join(root, file)
+                    write_path = os.path.relpath(file_path, dir_path)
+                    zip_file.write(file_path, write_path)
+
+
+def post_zip_to_server(tenant_name, username, password):
+    token = get_auth_token(username, password)
+    auth_header = {'Authorization': token}
+    post_data = {"tenantNameSuffix": tenant_name}
+    file_data = {'zipFile': open(ZIPFILE)}
+    url = base_url + '/configuration/deployTest'
+    r = requests.post(url, post_data, files=file_data,
+                      verify=False, headers=auth_header)
+    return r
+
+
+def get_auth_token(username, password):
+    post_data = {"username": username,
+                 "password": password}
+    auth_url = base_url + '/account/authenticate'
+    r = requests.post(auth_url, json=post_data, verify=False)
+    json_response = json.loads(r.text)
+    return json_response['authorizationToken']
+
+
+def parse_response(response, debug):
+    try:
+        json_response = json.loads(response.text)
+    except ValueError:
+        print 'An error happened!\n'
+        print 'Headers: %s' % response.headers
+        print 'Status Code: %s' % response.status_code
+        print 'Data: %s' % response.text
+        return
+    if debug is True:
+        print json_response
+        print response.text
+
+    if json_response.get('success'):
+        print "============== load was successful =============="
+        print json_response['hostname']
+    else:
+        print "============== load failed =============="
+
+
+if __name__ == '__main__':
+    args = parse_args()
+    create_zip_from_assets()
+    if args.username is None:
+        username = os.environ['SOCOTRA_USERNAME']
```

Comment:
Need to protect also for the case where environment variables do not exist...

Reply:
It only checks env variables if the command line arguments are null.   If there's no username on the command-line and no env variable set, it will fail which I think is the right behavior.

Reply:
Ok. But there are no instructions on why...

Reply:
Not really. Perhaps you can add a usage in the file or as a message, i.e., -h. I find the suffix only input confusing. Also, 'configeditor' is not a feasible suffix and you should point out.